lib/Doctrine/DBAL/Schema/Index.php

```diff
@@ -212,7 +214,7 @@ public function isFullfilledBy(Index $other)
                 return false;
             }
 
-            if (! $this->hasSameColumnLengths($other)) {
+            if (! $this->sameFlag(static::FLAG_SPATIAL, $other) && ! $this->hasSameColumnLengths($other)) {
```

Comment:
Why is only the `SPATIAL` flag of interest? Shouldn't it be something like `$this->hasSameFlags($other)`?

Reply:
We don't want to check other index flags. This is applicable for spatial only. I'm open to suggestions.

Reply:
Why don't we? Does it mean that if any other flag is added or removed to the index definition, we want to consider them equal and not cause a schema change?

Reply:
I'm also not a fan of `static::`. Is this currently necessary?